deeppavlov/skills/aiml_skill/aiml_skill.py

```diff
@@ -0,0 +1,139 @@
+from pathlib import Path
+from typing import Union
+from typing import Tuple, Optional, List
+from logging import getLogger
+import aiml
+import uuid
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.skill.skill import Skill
+log = getLogger(__name__)
+
+
+@register("aiml_skill")
+class AIMLSkill(Skill):
+    """Skill wraps python-aiml library into DeepPavlov interfrace.
+    AIML uses directory with AIML scripts which are loaded at initialization and used as patterns
+    for answering at each step.
+    """
+
+    def __init__(self,
+                 path_to_aiml_scripts: Union[str, None] = None,
+                 positive_confidence: float = 0.66,
+                 null_response: str = "I don't know",
+                 null_confidence: float = 0.33,
+                 **kwargs
+                 ) -> None:
+        """
+        Construct skill:
+            read AIML scripts,
+            load AIML kernel
+
+        Args:
+            path_to_aiml_scripts: string path to folder with AIML scripts
+            null_response: Response string to answer if no AIML Patterns matched
+            positive_confidence: The confidence of response if response was found in AIML scripts
+            null_confidence: The confidence when AIML scripts has no rule for responding and system returns null_response
+        """
+
+        if not path_to_aiml_scripts:
+            cur_dir = Path(__file__).absolute().parent
+            self.path_to_aiml_scripts = cur_dir.joinpath("aiml_scripts")
+            log.warning(f"path_to_aiml_scripts is not provided, use default path: `{self.path_to_aiml_scripts}`")
+        else:
+            self.path_to_aiml_scripts = Path(path_to_aiml_scripts)
+            log.info(f"path_to_aiml_scripts is: `{self.path_to_aiml_scripts}`")
+
+        self.positive_confidence = positive_confidence
+        self.null_confidence = null_confidence
+        self.null_response = null_response
+        self.kernel = aiml.Kernel()
+        # to block AIML output:
+        self.kernel._verboseMode = False
+        self._load_scripts()
+
+    def _load_scripts(self) -> None:
+        """
+        Scripts are loaded recursively from files with extensions .xml and .aiml
```

Comment:
I understand this just a wrapper, but would be nice to have an example of these .xml or .aiml files structure in the docstring.

Reply:
Or maybe a link to them?

Reply:
Really there is no requirements for file structure of AIML scripts. It is just a bunch of independent scripts which can be loaded to Kernel independently (but developer can inject dependencies into their scripts).

Examples of files are taken from here:
https://github.com/pandorabots/Free-AIML
(exactly I've taken only two scripts: `basic_chat.aiml`, `comedian.aiml`) and placed it into aiml_scripts folder.

This is described here:
https://github.com/deepmipt/DeepPavlov/blob/aiml_skill/deeppavlov/skills/aiml_skill/README.md
and here:
https://github.com/deepmipt/DeepPavlov/blob/aiml_skill/docs/skills/aiml_skill.rst